ui/main/src/app/modules/usercard/usercard.component.ts

```diff
@@ -250,25 +251,32 @@ export class UserCardComponent implements OnDestroy, OnInit {
             || (perimeter.rights === RightsEnum.Write));
     }
 
-    loadStatesForProcess(process: Process): void {
+    // return false if there is no states available in the process for writing card
+    loadStatesForProcess(process: Process): boolean {
         const statesList = [];
         this.currentUserWithPerimeters.computedPerimeters.forEach(
             perimeter => {
                 if ((perimeter.process === process.id) && UserCardComponent.userCanSendCard(perimeter)) {
-                    const state = process.states[perimeter.state];
-                    if (!!state) {
-                        if (!!state.userCard) {
-                            const label = !!state.name ? (new I18n(Utilities.getI18nPrefixFromProcess(process)
-                                + state.name)) : perimeter.state;
-                            const stateEntry = { value: perimeter.state, label: label };
-                            statesList.push(stateEntry);
-                        }
-                    } else console.log('WARNING : state', perimeter.state , 'is present in perimeter for process'
-                                    , process.id , 'but not in process definition');
-
+                    const state = this.getStateFromProcessDefinition(process,perimeter.state)
+                    if (!!state) statesList.push(state);
                 }
             });
-        this.statesPerProcesses.set(process.id, statesList);
+        if (statesList.length>0) this.statesPerProcesses.set(process.id, statesList);
```

Comment:
With this condition, if a process had states available for usercards, and then after an update it doesn't anymore, won't the statePerProcesses continue to contain the previous list (of states that are no longer avaiblable) as the temporary variable "stateList" will have a length of 0?